sdk/src/main/java/io/opentelemetry/sdk/trace/config/TraceParams.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.sdk.trace.config;
+
+import com.google.auto.value.AutoValue;
+import com.google.common.base.Preconditions;
+import io.opentelemetry.trace.Event;
+import io.opentelemetry.trace.Link;
+import io.opentelemetry.trace.Span;
+import javax.annotation.concurrent.Immutable;
+
+/** Class that holds global trace parameters. */
+@AutoValue
+@Immutable
+public abstract class TraceParams {
```

Comment:
We can name this TraceConfig if we go with my proposal. Also see all the fields in https://github.com/open-telemetry/opentelemetry-proto/blob/master/src/opentelemetry/proto/trace/v1/trace_config.proto#L37

Reply:
Sounds good, consolidated in https://github.com/open-telemetry/opentelemetry-java/pull/336/commits/b6c3859e78704d3dd802285855b78a0cf5407103.

Other configs are added in https://github.com/open-telemetry/opentelemetry-java/pull/336/commits/c8574082521b60015e5549aef63324b504070e1a.